provisioning/miniprov/hare_mp/utils.py

```diff
@@ -183,10 +185,59 @@ def is_component(self, machine_id: str, name: str) -> bool:
                 break
         return found
 
+    @func_log(func_enter, func_leave)
+    def is_component_services(self, machine_id: str,
+                              comp_name: str, svc_name: str) -> bool:
+        """
+        Returns True if the given component is present in the components
+        list for the given node>{machine_id} according to the
+        ConfStore (ValueProvider).
+        """
+        comp_names = self.provider.get(f'node>{machine_id}>'
+                                       f'components')
+        comp_present = self.is_component(machine_id, comp_name)
+
+        found: bool = False
+        if comp_present:
+            for component in comp_names:
+                svc_names = component.get('services')
+                if (svc_names is not None):
+                    for service in svc_names:
+                        if(service == svc_name):
+                            found = True
+                            break
+        return found
+
+    @func_log(func_enter, func_leave)
+    def is_client(self, machine_id: str,
+                  comp_name: str, client_name: str) -> bool:
+        """
+        Returns True if the given client is present in the components
+        list for the given node>{machine_id} according to the
+        ConfStore (ValueProvider).
+        """
+        comp_names = self.provider.get(f'node>{machine_id}>'
+                                       f'components')
+        comp_present = self.is_component(machine_id, comp_name)
+
+        found: bool = False
+        for component in comp_names:
+            svc_names = component.get('services')
+            if (svc_names is not None):
+                for service in svc_names:
+                    if(service == client_name):
+                        found = True
+                        break
+            else:
+                # TEMPORY: If services are present then it will check for
```

Comment:
Lets remove this and test by simulating final version of confstore

Reply:
Removed